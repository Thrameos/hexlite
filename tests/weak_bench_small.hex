% for debugging
%#maxint=2.
% as a testcase
#maxint=5.
% for benchmarking (2.5 GB RAM without --heuristics=monolithic)
%#maxint=10.

something(X) :- #int(X).

1 <= { foo(X) ; bar(X) ; baz(X) } <= 1 :- something(X).

nbr(N) :- N = #count { X : foo(X) }.
:~ nbr(N), NN=N+1. [NN:1]
