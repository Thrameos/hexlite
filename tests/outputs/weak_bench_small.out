{something(0),something(1),something(2),something(3),something(4),something(5),baz(0),baz(1),baz(2),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(5),baz(0),baz(1),baz(2),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(4),baz(0),baz(1),baz(2),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(4),bar(5),baz(0),baz(1),baz(2),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(3),baz(0),baz(1),baz(2),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(3),bar(5),baz(0),baz(1),baz(2),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(3),bar(4),baz(0),baz(1),baz(2),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(3),bar(4),bar(5),baz(0),baz(1),baz(2),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),baz(0),baz(1),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(5),baz(0),baz(1),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(4),baz(0),baz(1),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(4),bar(5),baz(0),baz(1),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(3),baz(0),baz(1),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(3),bar(5),baz(0),baz(1),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(3),bar(4),baz(0),baz(1),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(2),bar(3),bar(4),bar(5),baz(0),baz(1),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),baz(0),baz(2),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(5),baz(0),baz(2),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(4),baz(0),baz(2),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(4),bar(5),baz(0),baz(2),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(3),baz(0),baz(2),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(3),bar(5),baz(0),baz(2),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(3),bar(4),baz(0),baz(2),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(3),bar(4),bar(5),baz(0),baz(2),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),baz(0),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(5),baz(0),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(4),baz(0),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(4),bar(5),baz(0),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(3),baz(0),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(3),bar(5),baz(0),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(3),bar(4),baz(0),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(1),bar(2),bar(3),bar(4),bar(5),baz(0),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),baz(1),baz(2),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(5),baz(1),baz(2),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(4),baz(1),baz(2),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(4),bar(5),baz(1),baz(2),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(3),baz(1),baz(2),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(3),bar(5),baz(1),baz(2),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(3),bar(4),baz(1),baz(2),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(3),bar(4),bar(5),baz(1),baz(2),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),baz(1),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(5),baz(1),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(4),baz(1),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(4),bar(5),baz(1),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(3),baz(1),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(3),bar(5),baz(1),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(3),bar(4),baz(1),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(2),bar(3),bar(4),bar(5),baz(1),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),baz(2),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(5),baz(2),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(4),baz(2),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(4),bar(5),baz(2),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(3),baz(2),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(3),bar(5),baz(2),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(3),bar(4),baz(2),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(3),bar(4),bar(5),baz(2),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),baz(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(5),baz(3),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(4),baz(3),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(4),bar(5),baz(3),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(3),baz(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(3),bar(5),baz(4),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(3),bar(4),baz(5),nbr(0)} <[1:1]>
{something(0),something(1),something(2),something(3),something(4),something(5),bar(0),bar(1),bar(2),bar(3),bar(4),bar(5),nbr(0)} <[1:1]>
